pkg/statistics/handle/usage/collector/collector.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"sync"
+	"time"
+)
+
+// GlobalCollector provides a utility to collect stats data from each session
+type GlobalCollector[T any] interface {
+	SpawnSession() SessionCollector[T]
+	Close()
+}
+
+var _ GlobalCollector[int] = &globalCollector[int]{}
+
+// globalCollector is an implementation of `GlobalCollector`
+type globalCollector[T any] struct {
+	mergeFn            func(T)
+	dataCh             chan T
+	highPriorityDataCh chan T
+	closeCh            chan struct{}
+	wg                 sync.WaitGroup
+	timeout            time.Duration
+
+	closeOnce sync.Once
+}
+
+// NewGlobalCollector creates a new global collector
+func NewGlobalCollector[T any](mergeFn func(T), timeout time.Duration, channelSize int) GlobalCollector[T] {
+	g := &globalCollector[T]{
+		mergeFn:            mergeFn,
+		timeout:            timeout,
+		dataCh:             make(chan T, channelSize),
+		highPriorityDataCh: make(chan T, channelSize),
+		closeCh:            make(chan struct{}),
+	}
+	g.startWorker()
```

Comment:
Just a suggestion that we introduce a separate `Start()` method to it instead of starting it here implicitly.

Reply:
An explicit `Start()` seems safer, which can notify the caller that it will spawn a background goroutine.

Reply:
Done.